ui/src/pages/version/HistoryDetails.js

```diff
@@ -14,48 +14,50 @@ import { useMerlinApi } from "../../hooks/useMerlinApi";
 
 const defaultTextSize = "s";
 
-const DeploymentStatus = ({
-  status,
-  deployment,
-  endpointRevisionId,
-  endpointStatus,
-}) => {
-  if (status === "running" || status === "serving") {
-    if (
-      deployment.revision_id === endpointRevisionId &&
-      (endpointStatus === "pending" ||
-        endpointStatus === "running" ||
-        endpointStatus === "serving")
-    ) {
+const DeploymentStatus = ({ status }) => {
+  switch (status) {
+    case "pending":
+      return <EuiHealth color="gray">Pending</EuiHealth>;
+    case "running":
+    case "serving":
       return <EuiHealth color="success">Deployed</EuiHealth>;
-    }
-    return <EuiHealth color="default">Not Deployed</EuiHealth>;
-  } else if (status === "pending") {
-    return <EuiHealth color="gray">Pending</EuiHealth>;
-  }
-
-  if (deployment.error !== "") {
-    return <EuiHealth color="danger">Failed</EuiHealth>;
+    case "terminated":
+      return <EuiHealth color="default">Terminated</EuiHealth>;
+    case "failed":
+      return <EuiHealth color="danger">Failed</EuiHealth>;
+    default:
+      return <EuiHealth color="subdued">-</EuiHealth>;
   }
 };
 
 const RevisionPanel = ({ deployments, deploymentsLoaded, endpoint }) => {
   const [orderedDeployments, setOrderedDeployments] = useState([]);
+  const [currentDeployment, setCurrentDeployment] = useState({ id: null });
+
   useEffect(() => {
-    let orderedDeployments = deployments.sort((a, b) => (a.id > b.id ? 1 : -1));
-
-    let idx = 0;
-    orderedDeployments.forEach((deployment) => {
-      if (deployment.status === "running" || deployment.status === "serving") {
-        deployment.revision_id = idx + 1;
-        idx++;
-      }
+    if (deployments.length < 1) {
+      return;
+    }
+
+    // Sort deployments by id descending (newest first)
+    const ordered = deployments.sort((a, b) => (a.id < b.id ? 1 : -1));
+    setOrderedDeployments(ordered);
+
+    let currentDeployment = ordered.find((deployment) => {
+      return (
+        (deployment.status === "running" ||
+          deployment.status === "serving" ||
+          deployment.status === "terminated") &&
+        deployment.error === ""
+      );
     });
 
-    orderedDeployments = orderedDeployments.sort((a, b) =>
-      a.id < b.id ? 1 : -1
-    );
-    setOrderedDeployments(orderedDeployments);
+    // If no successful deployment, use the first deployment as current
```

Comment:
What if there is an active "pending" deployment? Should we pick the first "failed" deployment here?

So, in general, we have 3 cases:
* There is a serving / running / terminated deployment - pick the first one we find as current, by descending order of deployment id and use it (note - at most one of {serving, running} may be found)
* No serving / running / terminated deployment but there is a failed deployment - the latest one should be considered current
* No serving / running / terminated / failed deployments (both conditions above not satisfied) - this means, we only have a single deployment (we already checked `if (deployments.length < 1)` above) and it's in the pending state (guess we can skip the label for this).

Does it make sense?

Reply:
Yup, it makes sense. Thanks for the input!

For only Pending deployment:
<img width="500" alt="image" src="https://github.com/caraml-dev/merlin/assets/8122852/eefba72a-f678-4d3d-bf0d-dcded6495e6d">

Reply:
For Pending and Terminated:

<img width="500" alt="image" src="https://github.com/caraml-dev/merlin/assets/8122852/490d3767-fefa-4372-af39-e2fd56aee9c6">